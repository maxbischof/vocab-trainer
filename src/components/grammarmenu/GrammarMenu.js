import React from 'react'
import styled from 'styled-components'
import GrammarSubMenus from './GrammarSubMenus'
import { ReactComponent as CloseIcon } from '../../icons/delete.svg'
import WordClassMenu from './WordClassMenu'

export default function GrammarMenu({
  wordClass,
  setWordClass,
  gender,
  setGender,
  person,
  setPerson,
  number,
  setNumber,
  setShowGrammarMenu,
}) {
  return (
    <>
      <StyledGrammarMenu>
        <StyledCloseIcon onClick={() => setShowGrammarMenu(false)} />
        <GrammarSubMenus
          wordClass={wordClass}
          gender={gender}
          setGender={setGender}
          person={person}
          setPerson={setPerson}
          number={number}
          setNumber={setNumber}
        />
        <WordClassMenu wordClass={wordClass} setWordClass={setWordClass} />
      </StyledGrammarMenu>
    </>
  )
}

const StyledGrammarMenu = styled.div`
  display: grid;
  background: var(--background);
  grid-template-rows: 20px 1fr;
  grid-template-columns: 1fr auto;
  justify-items: center;
  align-items: center;
  position: absolute;
  padding: 20px 0 20px 0;
  width: 250px;
  height: 190px;
`
const StyledCloseIcon = styled(CloseIcon)`
  cursor: pointer;
  grid-row: 1/2;
  grid-column: 2/3;
`
