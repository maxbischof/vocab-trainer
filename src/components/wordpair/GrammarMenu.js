import React, { useState } from 'react'
import styled from 'styled-components'
import GrammarSubMenu from './GrammarSubMenu'
import Selection from './Selection'

export default function GrammarMenu({
  wordClass,
  setWordClass,
  gender,
  setGender,
  person,
  setPerson,
  number,
  setNumber,
}) {
  return (
    <>
      <GrammarSubMenu
        wordClass={wordClass}
        gender={gender}
        setGender={setGender}
        person={person}
        setPerson={setPerson}
        number={number}
        setNumber={setNumber}
      />
      <StyledGrammarMenu>
        <Selection
          text="N"
          activeProperty={wordClass}
          color="var(--noun)"
          diameter="63px"
          onClick={setWordClass}
        />

        <Selection
          text="A"
          activeProperty={wordClass}
          color="var(--adjective)"
          diameter="63px"
          onClick={setWordClass}
        />

        <Selection
          text="V"
          activeProperty={wordClass}
          color="var(--verb)"
          diameter="63px"
          onClick={setWordClass}
        />
      </StyledGrammarMenu>
    </>
  )
}

const StyledGrammarMenu = styled.div`
  display: flex;
  justify-content: space-around;
  width: 95vw;
`
