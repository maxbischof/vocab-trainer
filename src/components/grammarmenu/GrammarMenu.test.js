import React from 'react'
import GrammarMenu from './GrammarMenu'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<GrammarMenu />).toJSON()
  expect(tree).toMatchSnapshot()
})
