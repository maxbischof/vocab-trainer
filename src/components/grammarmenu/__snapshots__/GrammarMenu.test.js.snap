// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gsTCUz jBFAUf"
>
  <div
    className="sc-bdfBwQ klFbRb"
    color="var(--noun)"
    fontSize={31.5}
    onClick={[Function]}
  >
    N
  </div>
  <div
    className="sc-bdfBwQ cQXCMy"
    color="var(--adjective)"
    fontSize={31.5}
    onClick={[Function]}
  >
    A
  </div>
  <div
    className="sc-bdfBwQ emUPPW"
    color="var(--verb)"
    fontSize={31.5}
    onClick={[Function]}
  >
    V
  </div>
</div>
`;
